playbooks/robusta_playbooks/kubectl_enrichments.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import os
+import uuid
+
+from hikaru.model.rel_1_26 import Container, PodSpec
+from robusta.api import (
+    RUNNER_SERVICE_ACCOUNT,
+    ExecutionBaseEvent,
+    FileBlock,
+    MarkdownBlock,
+    PodRunningParams,
+    RobustaJob,
+    action,
+)
+from robusta.utils.parsing import format_event_templated_string
+
+IMAGE: str = os.getenv("KUBECTL_IMAGE_OVERRIDE", f"bitnami/kubectl:latest")
+
+
+class KubectlParams(PodRunningParams):
+    """
+    :var kubectl_command: The full kubectl command to run, formatted as a shell command string.
+    :var description: A description of the command ran.
+    :var timeout: The maximum time (in seconds) to wait for the kubectl command to complete. Default is 3600 seconds.
+    """
+
+    command: str = None  # type: ignore
+    description: str = None
+    timeout: int = 3600
+
+
+@action
+def kubectl_command(event: ExecutionBaseEvent, params: KubectlParams):
+    """
+    Runs a custom kubectl command inside a Kubernetes pod using a Job.
+    """
+
+    subject = event.get_subject()
+    formatted_kubectl_command = format_event_templated_string(subject, params.command)
+    
+    logging.debug(f"kubectl_command running '{formatted_kubectl_command}'")
+
+    spec = PodSpec(
+        serviceAccountName=RUNNER_SERVICE_ACCOUNT,
+        containers=[
+            Container(
+                name="kubectl",
+                image=IMAGE,
+                imagePullPolicy="Always",
+                command=["/bin/sh", "-c"],
+                args=[formatted_kubectl_command]
+            )
+        ],
+        restartPolicy="Never",
+    )
+
+    try:
+        kubectl_response = RobustaJob.run_simple_job_spec(
+            spec,
+            f"debug-kubectl-{str(uuid.uuid4())}",
+            params.timeout,
+            custom_annotations=params.custom_annotations,
+            ttl_seconds_after_finished=43200,  # 12 hours
```

Comment:
Can we add all those args to `KubectlParams`? I think it is good idea letting the user decide if he want to override those.

Reply:
A. I think that it is something that possible for users to want to change, don't remove the job after it run or change ttl.
B. It is small effort no? if no then ignore if yes let's just do it.
C. From my point of view it is very confusing we are passing ttl but we anyway delete the job when it is finished.

Reply:
opened ticket